swifttool/manager.py

```diff
@@ -0,0 +1,288 @@
+                        for dev in devices:
+                            devname = dev['device']
+                            if devname not in ring_devs:
+                                initial_weight = 0
+                            else:
+                                initial_weight = ring_devs[devname]
+                            if initial_weight == dev['target']:
+                                continue
+                            devinfo = {}
+                            if 'metadata' in dev:
+                                devinfo['metadata'] = dev['metadata']
+                            devinfo['blockdev'] = devname
+                            devinfo['weight'] = int(initial_weight +
+                                                    dev['step'])
+                            if abs(devinfo['weight'] - dev['target']) < 50:
+                                devinfo['weight'] = dev['target']
+                            LOG.debug("Setting the %s device capacity to %s",
+                                      devname, str(devinfo['weight']))
+                            newdevs.append(devinfo)
+                            changed = True
+                        updated_cfg['zones'][zone][ip]['disks'][builder] = \
+                            newdevs
+            if changed:
+                ringsdef_helper(updated_cfg, self.metadata, self.outdir)
+            return changed, updated_cfg
+        except Exception as e:
+            print "Error updating cluster configuration", e
+            sys.exit(-1)
+
+    @staticmethod
+    def _get_zone_ip(devs):
+        for dev in devs:
+            if 'ip' in dev:
+                return dev['ip']
+
+    def _calculate_cluster_changes(self):
+        try:
+            self.cluster_config = yaml.load(open(self.def_file, 'r'))
+            expected_config = get_devices(self.cluster_config['zones'])
+            current_config = self._get_all_cluster_devices()
+            # To get devices to be removed we look for devices present in
+            # current_config and not in expected_config
+            # To get list of devices to be added look for devices in expected
+            # not in current
+            if self.scaleup:
+                changes = self._get_devices_changed(expected_config,
+                                                    current_config)
+            else:
+                changes = self._get_devices_changed(current_config,
+                                                    expected_config)
+            self._normalize_changes(changes)
+            # Each ring will be list of devices to be modified
+            # Each device (in ring) will have following attributes:
+            # - zone
+            # - ip
+            # - device
+            # - target (device weight when adding, 0 when removing)
+            # - step (amount to change every time till target reached)
+            # step is defaulted to 10% or -10% of target
+        except Exception as e:
```

Comment:
Same here.  I'd not use this catch-all exception block here, and I'd bubble up the error directly on out.  Remove this entirely.  So much could go wrong in this block, and when it does, how will you ever figure out what the problem is without a traceback and a miracle?